src/test/feesutil_tests.cpp

```diff
@@ -0,0 +1,328 @@
+            BOOST_CHECK(descendants.size() == 1);
+        }
+    }
+    // Test 3 Linear transactions clusters
+    /*
+            Linear Packages
+            A     B     C    D
+            |     |     |    |
+            E     H     J    K
+            |     |
+            F     I
+            |
+            G
+    */
+    {
+        std::vector<RemovedMempoolTransactionInfo> transactions;
+        transactions.reserve(11);
+        // Create transaction A B C D
+        for (auto i = 0; i < 4; ++i) {
+            const std::vector<COutPoint> outpoints{COutPoint(Txid::FromUint256(InsecureRand256()), 0)};
+            const CTransactionRef tx = make_tx(outpoints);
+            transactions.emplace_back(entry.FromTx(tx));
+        }
+
+        // Create cluster A children ---> E->F->G
+        std::vector<COutPoint> outpoints{COutPoint(transactions[0].info.m_tx->GetHash(), 0)};
+        for (auto i = 0; i < 3; ++i) {
+            const CTransactionRef tx = make_tx(outpoints);
+            transactions.emplace_back(entry.FromTx(tx));
+            outpoints = {COutPoint(tx->GetHash(), 0)};
+        }
+
+        // Create cluster B children ---> H->I
+        outpoints = {COutPoint(transactions[1].info.m_tx->GetHash(), 0)};
+        for (size_t i = 0; i < 2; ++i) {
+            const CTransactionRef tx = make_tx(outpoints);
+            transactions.emplace_back(entry.FromTx(tx));
+            outpoints = {COutPoint(tx->GetHash(), 0)};
+        }
+
+        // Create cluster C child ---> J
+        outpoints = {COutPoint(transactions[2].info.m_tx->GetHash(), 0)};
+        const CTransactionRef txJ = make_tx(outpoints);
+        transactions.emplace_back(entry.FromTx(txJ));
+
+        // Create cluster B child ---> K
+        outpoints = {COutPoint(transactions[3].info.m_tx->GetHash(), 0)};
+        const CTransactionRef txK = make_tx(outpoints);
+        transactions.emplace_back(entry.FromTx(txK));
+        const auto txAncestorsAndDescendants = GetTxAncestorsAndDescendants(transactions);
+        BOOST_CHECK_EQUAL(txAncestorsAndDescendants.size(), transactions.size());
+        // Check tx A topology;
+        {
+            const Txid txA_Id = transactions[0].info.m_tx->GetHash();
+            const auto txA_Iter = txAncestorsAndDescendants.find(txA_Id);
+            BOOST_CHECK(txA_Iter != txAncestorsAndDescendants.end());
+            const auto ancestors = std::get<0>(txA_Iter->second);
+            BOOST_CHECK(ancestors.size() == 1); // A
+            BOOST_CHECK(ancestors.find(txA_Id) != ancestors.end());
+            const auto descendants = std::get<1>(txA_Iter->second);
+            BOOST_CHECK(descendants.size() == 4); // A, E, F, G
+            BOOST_CHECK(descendants.find(txA_Id) != descendants.end());
+            for (auto i = 4; i <= 6; i++) {
+                auto curr_tx = transactions[i];
+                BOOST_CHECK(descendants.find(curr_tx.info.m_tx->GetHash()) != descendants.end());
+            }
+        }
```

Comment:
There are 9 such assertion blocks with almost similar validations on ancestors and descendants.
If the author feels it's worth it, IMO this duplication can be eliminated by accepting few variables as the input in a function and calling it with different arguments.
```
validateTxRelations(txToValidate, ancestorTxs, descendantTxs)
```

Reply:
thanks accepted your suggestion with a slight modification.

Reply:
Great, thank you!